src/Plugins/OrientationAnalysis/src/OrientationAnalysis/Filters/Algorithms/BadDataNeighborOrientationCheck.cpp

```diff
@@ -81,16 +74,12 @@ Result<> BadDataNeighborOrientationCheck::operator()()
   std::vector<int32_t> neighborCount(totalPoints, 0);
 
   int64_t progressInt = 0;
-  auto start = std::chrono::steady_clock::now();
+
   for(size_t i = 0; i < totalPoints; i++)
   {
-    auto now = std::chrono::steady_clock::now();
-    if(std::chrono::duration_cast<std::chrono::milliseconds>(now - start).count() > 1000)
+    if(shouldSend())
```

Comment:
`sendMessage` calls `shouldSend()` already so you don't need this extra check. If you want to avoid having to call `fmt::format()` if a message isn't sent, then you can use the overload that takes a functor: e.g. `sendMessage([progressInt](){ return fmt::format("Processing Data '{}'% completed", progressInt); })`

This isn't the only spot for this as well.